### PR TITLE
Provide bounce option

### PR DESCRIPTION
**Pull request by netil**
## Issue
#168 

## Details
Added to handle bounce option
Separated init settings to _setOptions() and _setConfig() methods
Changed _getDataByDirection() to not alter param value

## Preferred reviewers
@sculove @happyhj

### CODE REVIEW

**Comment by sculove** [comment]
LGTM

File: src/flicking.js [hunk 1]
```diff
@@ -101,27 +102,72 @@ eg.module("flicking", ["jQuery", eg, window, document, eg.MovableCoord], functio
...
+		_setOptions: function(options) {
+			var padding = [0, 0];
+			var bounce = [10, 10];
+			var self = this;
+
+			var setDefault = function (name, defVal) {
+				var options = self.options;
+				var val = options[ name ];
+
+				if (typeof val === "number") {
```
Comment (mixed) [inline src/flicking.js]: Could you use `isNumeric`?

File: src/flicking.js [hunk 2]
```diff
@@ -101,27 +102,72 @@ eg.module("flicking", ["jQuery", eg, window, document, eg.MovableCoord], functio
...
+					val = $.each(defVal, function(i) {
+						defVal[i] = val;
+					});
+				} else if (val.constructor !== Array) {
```
Comment (mixed) [inline src/flicking.js]: and `isArray`.

File: src/flicking.js [hunk 3]
```diff
@@ -101,27 +102,72 @@ eg.module("flicking", ["jQuery", eg, window, document, eg.MovableCoord], functio
...
+		/**
+		 * Set options values
+		 * @param {Object} options
+		 */
+		_setOptions: function(options) {
+			var padding = [0, 0];
+			var bounce = [10, 10];
+			var self = this;
+
+			var setDefault = function (name, defVal) {
```
Comment (mixed) [inline src/flicking.js]: I think that it is ambiguous function name. `setDefault` is general name but It is not general working.

**Comment by mixed** [comment]
👍

**Comment by mixed** [comment]
👍

**Pull request closed without merging**